src/plugins/kibana_utils/public/state_management/url/kbn_url_storage.ts

```diff
@@ -238,7 +238,8 @@ export const createKbnUrlControls = (
  * 4. Hash history with base path
  */
 export function getRelativeToHistoryPath(absoluteUrl: string, history: History): History.Path {
-  function stripBasename(path: string = '') {
+  function stripBasename(path: string | null) {
```

Comment:
Mind shedding some light on the reasoning behind changing this behavior? What was the edge case here?

Reply:
If I read the code correctly, the `path` argument can never be `undefined` because it's given the value of `url.parse(x).pathname`, and `pathname` is either a string or `null`. So the default value (an empty string) was never set here in case `path` was `null`.

Now that I say that I can see that the change to the line below where this function is called isn't really necessary as `pathname` as I just said will never be `undefined`. So the fallback to `null` is redundant. Don't you agree?

Reply:
Ah, now I remember why I needed the `?? null` in the call to `stripBasename` below. If I don't add that I get this TypeScript error when running the type check for `x-pack/examples/alerting_example`:

![image](https://user-images.githubusercontent.com/10602/98662479-1f9e0800-2348-11eb-9451-5511c01c67ae.png)

I'm not really sure why it thinks that `pathname` can be `undefined` because the regular TypeScript tests for `kibana` doesn't seem to think so. And If I hover over `pathname` inside VSCode it also thinks that it can be `string | null`:

![image](https://user-images.githubusercontent.com/10602/98662600-51af6a00-2348-11eb-8b0e-11b41afe03c3.png)

So somehow the TypeScript tests in `x-pack/examples/alerting_example` is different. I'm not an expert of how these tests are running and if they might be running with a slightly different setup 🤷‍♂️ So I'm tempted to just leave it as is unless someone knows more than I.